app/assets/javascripts/host_edit.js

```diff
@@ -591,6 +591,27 @@ $(document).on('change', '.interface_type', function () {
   interface_type_selected(this);
 });
 
+$(document).on('change', '.host_name', function (event) {
```

Comment:
Based on the fact that the Foreman UI tries to slowly move away from the jQuery code, I think it would would save ourselves some work just by not introducing this here right now. For the user, I hope the more desriptive message, that you already did on the model, should be sufficient.